File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int, int> idx_map;
        int size = nums.size();
        for(int i = 0; i < size; i++){
            int compliment = target - nums[i];
            if(idx_map.find(compliment) != idx_map.end()){
                return {idx_map[compliment], i};
            }

            idx_map[nums[i]] = i;
        }

        return {};
        
        }

};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mymap = new HashMap<>();

        for(int i = 0 ; i < nums.length ; i++){
            
            if(mymap.containsKey(nums[i])){
                return new int[] {mymap.get(nums[i]) , i};
            }else{
                mymap.put((target - nums[i]), i);
            }
        }

        return new int[] {};
    }
}